pkg/api/types.go

```diff
@@ -398,6 +402,21 @@ type NFSVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// GlusterfsVolumeSource represents a Glusterfs Mount that lasts the lifetime of a pod
+type GlusterfsVolumeSource struct {
+	// Required: hosts is the endpoint name that details Glusterfs topology
+	Hosts string `json:"hosts"`
+
+	// Required: Path is the Glusterfs volume path
+	Path string `json:"path"`
+
+	// Optional: mountOptions is the mount time options such as "rw" or "ro"
+	MountOpt string `json:"mountOptions,omitempty"`
```

Comment:
I think figuring out which options is safe is a massive project.  But we should not expose unsafe ones in our first step.  Readonly is obviously safe.  So, lets just have `Readonly bool`.